system/src/system_control_internal.cpp

```diff
@@ -389,6 +399,19 @@ void system_ctrl_set_result(ctrl_request* req, int result, ctrl_completion_handl
     particle::system::SystemControl::instance()->setResult(req, result, handler, data);
 }
 
+int system_set_control_request_filter(uint16_t reqType, void* reserved) {
+    if (std::find(particle::system::SystemControl::vecCtrlReq.begin(), particle::system::SystemControl::vecCtrlReq.end(), reqType) == particle::system::SystemControl::vecCtrlReq.end()) {
+        LOG_DEBUG(TRACE, "Adding a new req to control req filter: %u", reqType);
+        particle::system::SystemControl::vecCtrlReq.insert(particle::system::SystemControl::vecCtrlReq.size(),reqType);
```

Comment:
Simply use `append()` and check the return value. If it failed to append, which means that there is probably no heap available, the return `SYSTEM_ERROR_NO_MEMORY`